code/mission/missionparse.cpp

```diff
@@ -2489,6 +2489,26 @@ void parse_bring_in_docked_wing(p_object *p_objp, int wingnum, int shipnum)
 	// make sure we haven't created too many ships
 	Assert(wingp->current_count <= MAX_SHIPS_PER_WING);
 
+	//fixes bug where docked wings are not called with the 'On Ship Arrive' hook variable --wookieejedi
```

Comment:
I see several problems with this PR.

1) `num_ships` will always be >0 here.
2) A wing which arrives because all its ships are docked will have no arrival anchor, so the check is unnecessary.
3) Your code loops through all the ships in the wing, which is wrong here because this function is already called for *each* ship in the wing, *as* it is created.
4) This function is the wrong place for the scripting hook -- normally, the hook is executed at the very bottom of `parse_create_object_sub`.  This is because the FREDder may need to access wing attributes that are assigned later in the function.  Instead of adding a hook to this function, you need to move this code to the same place as the other hook.

Reply:
Perhaps something like this at the end of `parse_create_object_sub`?

```
	// The ship is in a wing, but is docked. Adding this block ensures that docked ships actually execute "On Ship Arrive". --wookieejedi
	if (Game_mode & GM_IN_MISSION && shipp->wingnum != -1 && object_is_docked(pobjp)) {
		Script_system.SetHookObjects(2, "Ship", &Objects[objnum], "Parent", nullptr);
		Script_system.RunCondition(CHA_ONSHIPARRIVE, &Objects[objnum]);
		Script_system.RemHookVars(2, "Ship", "Parent");
	}	
```

Reply:
It's not possible to add the hook to `mission_set_wing_arrival_location()` because of the inside-out way that docked wings are initialized, so the hook will need to be added to the bottom of `parse_create_object_sub()`.

Your proposed code is partially correct, but it doesn't handle the correct set of scenarios.  There are some very specific conditions for initializing a docked ship in this way, and they are all handled in the check near the beginning of the function.

In fact, now that I think about it, the best solution is probably to create a new boolean variable `brought_in_docked_wing` and add this to the if() block at the end of `parse_create_object_sub()`.